src/nvim/api/buffer.c

```diff
@@ -745,10 +758,34 @@ void nvim_buf_set_name(Buffer buffer, String name, Error *err)
   }
 }
 
-/// Checks if a buffer is valid
+/// Checks if a buffer is valid and loaded. When a buffer is
+/// unloaded its file contents are freed from memory and vim
+/// cannot operate on the buffer lines until it is reloaded
+/// (usually by opening the buffer again in a new window). API
+/// methods such as |nvim_buf_line_count()| and
+/// |nvim_buf_get_lines()| will be affected.
+///
+/// Buffers may have been unloaded by the |:bunload| command or
+/// the buffer's|'bufhidden'| option.
+///
+/// @param buffer Buffer handle
+/// @return true if the buffer is valid and loaded, false otherwise.
+Boolean nvim_buf_is_loaded(Buffer buffer)
+  FUNC_API_SINCE(5)
+{
+  Error stub = ERROR_INIT;
```

Comment:
Maybe we should use `Error *err` here, then it would be possible to identify the three states: loaded, unloaded (thus loadable) and invalid in one RPC call.

Reply:
My hesitation with returning an error for invalid buffers is that you'd now need to introduce error handling code (try/except in python or pcall() in lua) to use buf.is_loaded(). Which in turn means you can't write a one-liner like `if buf.is_loaded() && buf.line_count() > 10:`

Reply:
That makes sense. We could add some more detailed state function later it need be.

Reply:
(for posterity: just realized `nvim_buf_line_count` is already this function.)